torch/_dynamo/variables/builder.py

```diff
@@ -1186,6 +1186,7 @@ def wrap_to_fake_tensor_and_record(
             for i in range(e.dim()):
                 # NB: mark dynamic has precedence over static
                 marked_dynamic = i in getattr(e, "_dynamo_dynamic_indices", set())
+                marked_weak_dynamic = i in getattr(e, "_dynamo_weak_dynamic_indices", set())
```

Comment:
Just to learn, what does `weak` mean here?

Reply:
Try to make it dynamic, but if it doesn't succeed don't complain